### PR TITLE
Fix: read_meter shortcode not displaying on individual page

### PR DESCRIPTION
**Pull request by osk02**
Fixed the [read-meter] short code not loading if post types are not selected. No duplicate read time blocks will be displayed repeatedly even after selecting a post type from plugin's general settings.

### CODE REVIEW

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: classes/class-bsfrt-readtime.php [hunk 1]
```diff
@@ -183,19 +183,25 @@ public function bsf_rt_get_option( $key, $default = false ) {
 	 * Frontend settings.
 	 */
 	public function bsf_rt_init_frontend() {
+		global $post;
+		if ( ! empty($post) && has_shortcode($post->post_content, 'read_meter') ) {
+			$this->bsf_rt_add_default_frontend_css();
+			$this->bsf_rt_add_readtime_styles_content();
+			return;
```
[the remark this answers is not in the document]
Comment (osk02) [reply classes/class-bsfrt-readtime.php:190]: Understood. The motive behind return was to ensure that the styles are not being loaded again even if post type was selected from the settings.

File: classes/class-bsfrt-readtime.php [hunk 2]
```diff
@@ -212,7 +225,7 @@ public function bsf_rt_init_frontend() {
 
 		// Show Reading time Conditions.
 		if ( $this->bsf_rt_get_option( 'bsf_rt_show_read_time' ) && 'none' !== $this->bsf_rt_options['bsf_rt_position_of_read_time'] ) {
-			if ( in_array( 'bsf_rt_single_page', $this->bsf_rt_options['bsf_rt_show_read_time'] ) && is_singular() ) {//PHPCS:ignore:WordPress.PHP.StrictInArray.MissingTrueStrict
+			if ( in_array( 'bsf_rt_single_page', $this->bsf_rt_options['bsf_rt_show_read_time'] ) && is_singular() && ! $this->bsf_rt_styles_loaded_flag ) {//PHPCS:ignore:WordPress.PHP.StrictInArray.MissingTrueStrict
```
Comment (RajkiranBagal) [inline classes/class-bsfrt-readtime.php:228]: I think for now don't handle duplicate reading time rendering, As this condition has above & below title rendering too, there might be some user's who might me using it.

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: classes/class-bsfrt-readtime.php [hunk 3]
```diff
@@ -183,25 +190,31 @@ public function bsf_rt_get_option( $key, $default = false ) {
...
 
 		if ( false === $this->bsf_rt_check_selected_post_types() ) {
 
 			return;
 		}
-		add_action( 'wp_enqueue_scripts', array( $this, 'bsfrt_frontend_default_css' ) );
+		$this->bsf_rt_add_default_frontend_css();
 		add_filter( 'comments_template', array( $this, 'bsf_rt_marker_for_progressbar' ) );
-
+
```
Comment (RajkiranBagal) [inline classes/class-bsfrt-readtime.php:206]: Remove these unwanted changes from below too

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: classes/class-bsfrt-readtime.php (hunk 3 above)
Comment (osk02) [reply classes/class-bsfrt-readtime.php:206]: Which changes are you referring to here?

Comment (RajkiranBagal) [reply classes/class-bsfrt-readtime.php:206]: Empty spaces

File: classes/class-bsfrt-readtime.php [hunk 4]
```diff
@@ -183,19 +190,25 @@ public function bsf_rt_get_option( $key, $default = false ) {
 	 * Frontend settings.
 	 */
 	public function bsf_rt_init_frontend() {
+		global $post;
+		if ( ! empty($post) && has_shortcode($post->post_content, 'read_meter') ) {
```
Comment (RajkiranBagal) [inline classes/class-bsfrt-readtime.php:194]:
```suggestion
		if ( ! empty( $post ) && has_shortcode( $post->post_content, 
 'read_meter' ) ) {
```

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**